ktoml-source/src/jvmTest/kotlin/com/akuleshov7/ktoml/source/StreamTests.kt

```diff
@@ -0,0 +1,219 @@
+            }
+        assertEquals(
+            listOf("a", "a.b.c", "a.d", "d", "d.a"),
+            parsedResult.getRealTomlTables().map { it.fullTableName })
+    }
+
+    @Serializable
+    data class RegressionTest(val a: Long?, val b: Long, val c: Long, val d: Long?)
+
+    @ExperimentalSerializationApi
+    @Test
+    fun regressionCast2Test() {
+        val parsedResult =
+            Toml().decodeFromStream<RegressionTest>(getTestDataStream("class_cast_regression2.toml"))
+        assertEquals(RegressionTest(null, 1, 2, null), parsedResult)
+    }
+
+    @ExperimentalSerializationApi
+    @Test
+    fun regressionPartialTest() {
+        val parsedResult =
+            Toml().decodeFromStream<RegressionTest>(getTestDataStream("class_cast_regression2.toml"))
+        assertEquals(RegressionTest(null, 1, 2, null), parsedResult)
+    }
+
+
+    @Serializable
+    data class TestRegression(
+        val list1: List<Double>,
+        val general: GeneralConfig,
+        val list2: List<Long>,
+        val warn: WarnConfig,
+        val list3: List<String>
+    )
+
+    @Serializable
+    data class GeneralConfig(
+        val execCmd: String? = null,
+        val tags: List<String>? = null,
+        val description: String? = null,
+        val suiteName: String? = null,
+        val excludedTests: List<String>? = null,
+        val includedTests: List<String>? = null,
+        val ignoreSaveComments: Boolean? = null
+    )
+
+    @Serializable
+    data class WarnConfig(
+        val list: List<String>
+    )
+
+    @ExperimentalSerializationApi
+    @Test
+    fun regressionInvalidIndex() {
+        assertEquals(
+            GeneralConfig(
+                execCmd = "echo hello world",
+                tags = listOf("Tag", "Other tag"),
+                description = "My description",
+                suiteName = "// DocsCheck",
+                excludedTests = null,
+                includedTests = null,
+                ignoreSaveComments = null
+            ),
+            Toml().partiallyDecodeFromStream(
+                getTestDataStream("partial_parser_regression.toml"),
+                "general"
+            )
+        )
+        assertEquals(
+            TestRegression(
+                list1 = listOf(1.0, 2.0),
+                general = GeneralConfig(
+                    execCmd = "echo hello world",
+                    tags = listOf("Tag", "Other tag"),
+                    description = "My description",
+                    suiteName = "// DocsCheck",
+                    excludedTests = null,
+                    includedTests = null,
+                    ignoreSaveComments = null
+                ),
+                list2 = listOf(1, 3, 5),
+                warn = WarnConfig(list = listOf("12a", "12f")),
+                list3 = listOf("mystr", "2", "3")
+            ),
+            Toml().decodeFromStream(getTestDataStream("partial_parser_regression.toml"))
```

Comment:
Better to have same syntax as we have for a straight forward decodeFromString:
`Toml.decodeFromString<Bool>("a = \"test\"")`

With a default object companion 

`@ThreadLocal
    public companion object Default : Toml(TomlConfig())`